File: src/main/resources/static/js/user/modules/mypageApi.js


export function getMypageZipList(page, callback) {
    fetch(`/api/v1/mypage/zips?page=${page}`)
        .then(resp => {
            if (resp.ok) {
                return resp.json();
            }
        })
        .then(data => {
            if (callback) {
                callback(data);
            }
        });
}


export function getMypageDnSellList(page, callback) {
    fetch(`/api/v1/mypage/dn/sell?page=${page}`)
        .then(resp => {
            if(resp.ok){
                return resp.json();
            }
        })
        .then(data => {
            if (callback) {
                callback(data);
            }
        })
}

export function patchMypageDnSell(dnId, callback) {
    fetch(`/api/v1/mypage/dn/sell/${dnId}`, {
        method: 'PATCH',
    })
        .then(resp => {
            if (resp.ok && callback) {
                callback();
            }
        });
}

export function getMypageHelpList(page, callback) {
    fetch(`/api/v1/mypage/main?page=${page}`)
        .then(resp => {
            if (resp.ok) {
                return resp.json();
            }
        })
        .then(data => {
            if (callback) {
                callback(data);
            }
        });
}

File: src/main/java/com/example/dncompany/controller/user/MypageApi.java
package com.example.dncompany.controller.user;

import com.example.dncompany.dto.page.PageDTO;
import com.example.dncompany.dto.page.PageRequestDTO;
import com.example.dncompany.dto.user.mypage.MypageDnBuyListDTO;
import com.example.dncompany.dto.user.mypage.MypageDnSellListDTO;
import com.example.dncompany.dto.user.mypage.MypageZipBoardListDTO;
import com.example.dncompany.service.user.MypageService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.*;

@Slf4j
@RestController
@RequiredArgsConstructor
@RequestMapping("/api")
public class MypageApi {
    private final MypageService mypageService;

    @GetMapping("/v1/mypage/zips")
    public PageDTO<MypageZipBoardListDTO> getMypageZips(PageRequestDTO pageRequestDTO,
                                @SessionAttribute("usersId") Long usersId) {
        PageDTO<MypageZipBoardListDTO> mypageZipBoardListDTOPageDTO = mypageService.zipBoardListPage(usersId, pageRequestDTO);
        return mypageZipBoardListDTOPageDTO;
    }

//    @GetMapping("/v1/mypage/dn/buy")
//    public PageDTO<MypageDnBuyListDTO> getMypageDnBuy(PageRequestDTO pageRequestDTO,
//                                                        @SessionAttribute("usersId") Long usersId){
//        PageDTO<MypageDnBuyListDTO> pageDTO = mypageService.mypageDnBuyListPage(usersId, pageRequestDTO);
//        return pageDTO;
//    }

    @GetMapping("/v1/mypage/dn/sell")
    public PageDTO<MypageDnSellListDTO> getMypageDnSell(PageRequestDTO pageRequestDTO,
                                                          @SessionAttribute("usersId") Long usersId){
        PageDTO<MypageDnSellListDTO> pageDTO = mypageService.mypageDnSellListPage(usersId, pageRequestDTO);
        return pageDTO;
    }

    @PatchMapping("/v1/mypage/dn/sell/{dnId}")
    public void patchMypageDnSell(@PathVariable Long dnId) {
        mypageService.modifyDnStatusByDnId(dnId);
    }

}
